File: Formula de Bashkara.py
#Formula general.
import math

print("Sea ax^2 + bx + c, con a distinto de cero. Ingrese los valores: ")
a = float(input("Introduzca a la variable a: "))
b = float(input("Introduzca a la variable b: "))
c = float(input("Introduzca a la variable c: "))

d = b**2 - 4*a*c

if d > 0:
    x1 = (-b + math.sqrt(d))/2*a
    print("x_1 = ", x1)
    x2 = (-b - math.sqrt(d))/2*a
    print("x_2 = ", x2)
elif d==0:
    x = -b/2*a
    print("x = ", x)
else:
    print("¡Ingresa valores Reales!")

File: Formula de Bashkara.f90
program formula_general
    implicit none
    real :: a, b, c, d
    real :: x1, x2

    ! Pedir al usuario que ingrese los valores de a, b, y c
    print*, "Sea ax^2 + bx + c, con a distinto de cero. Ingrese los valores: "
    print*, "Introduzca el valor de a: "
    read(*,*) a
    print*, "Introduzca el valor de b: "
    read(*,*) b
    print*, "Introduzca el valor de c: "
    read(*,*) c

    ! Calcular el discriminante
    d = b**2 - 4.0*a*c

    if (d > 0.0) then
        ! Dos soluciones reales distintas
        x1 = (-b + sqrt(d)) / (2.0 * a)
        x2 = (-b - sqrt(d)) / (2.0 * a)
        print*, "x_1 = ", x1
        print*, "x_2 = ", x2
    else if (d == 0.0) then
        ! Una solución real
        x1 = -b / (2.0 * a)
        print*, "x = ", x1
    else
        ! No hay soluciones reales
        print*, "¡No hay soluciones reales para los valores ingresados!"
    end if

end program formula_general
